.github/workflows/ci.yml

```diff
@@ -108,11 +108,29 @@ jobs:
           choco install ninja OpenCppCoverage -y
           echo "C:\Program Files\OpenCppCoverage" >> $env:GITHUB_PATH
 
+    - name: Cache MinGW
+      if: matrix.platform.name == 'Windows MinGW'
+      id: mingw-cache
+      uses: actions/cache@v3
+      with:
+        path: |
+          mingw64
+        key: winlibs-x86_64-posix-seh-gcc-12.2.0-llvm-16.0.0-mingw-w64msvcrt-10.0.0-r5
+
     - name: Install MinGW
+      if: matrix.platform.name == 'Windows MinGW' && steps.mingw-cache.outputs.cache-hit != 'true'
+      shell: bash
+      run: |
+        curl -Lo "$GITHUB_WORKSPACE/mingw64.zip" https://github.com/brechtsanders/winlibs_mingw/releases/download/12.2.0-16.0.0-10.0.0-msvcrt-r5/winlibs-x86_64-posix-seh-gcc-12.2.0-llvm-16.0.0-mingw-w64msvcrt-10.0.0-r5.zip
+        unzip -qq -d $GITHUB_WORKSPACE mingw64.zip
+
+    - name: Setup MinGW
       if: matrix.platform.name == 'Windows MinGW'
+      shell: bash
       run: |
-        choco uninstall mingw
-        choco install mingw --version 11.2.0
+        echo "CC=$GITHUB_WORKSPACE\\mingw64\\bin\\gcc.exe" >> $GITHUB_ENV
+        echo "CXX=$GITHUB_WORKSPACE\\mingw64\\bin\\g++.exe" >> $GITHUB_ENV
+        echo "RC=$GITHUB_WORKSPACE\\mingw64\\bin\\windres.exe" >> $GITHUB_ENV
```

Comment:
Why add this extra step instead of just changing the CMAKE_CXX_COMPILER configure param? I'd prefer we not not modify the environment when non-envvar options exist.

Reply:
This seemed like the cleaner of [both endorsed possibilities](https://gitlab.kitware.com/cmake/community/-/wikis/FAQ#how-do-i-use-a-different-compiler) considering we specify the location where we unpack MinGW in the step itself. I'd rather not have to think of multiple places to have to make adjustments if we decide to move the directory at some point.

Reply:
I think it's cleaner to simply change existing arguments than add a whole new CI step that requires another 8 lines of code, plus it's consistent with all other jobs which specify the compiler this way. I'd rather not mix and match how we specify such things.

I know in your mega-CI-rewrite PR you made heavy use of environment variables but I hope we don't go down that route. I'd rather specify things via existing CMake interfaces wherever possible. I want that because when you look at the [Configure CMake](https://github.com/SFML/SFML/actions/runs/6387165344/job/17335087763) CI step you can see all configure parameters listed out whereas environment variables remain invisible.